languages/java/oso/src/main/java/com/osohq/oso/Query.java

```diff
@@ -71,6 +71,10 @@ private void handleCall(
       ffiQuery.applicationError(e.getMessage());
       ffiQuery.callResult(callId, null);
       return;
+    } catch (Exceptions.InvalidAttributeError e) {
```

Comment:
This was causing uncaught runtime exceptions when specializing on non-existing fields.

E.g. `x matches { non_existent: 1 }` 

We rely on the `check_errors` flag to determine whether we want this to be a hard failure.

Incidentally... maybe we _do_ want this to be a hard error if specializing on a type. E.g. `x matches Foo { non_existent: 1}` should fail if `non_existent` doesn't exist. But not for dicts?

Reply:
Maybe it would make sense to match the semantics of the host language to determine what type of error it should be? E.g.,

```js
> class Foo {}
undefined
> new Foo().foo
undefined
```

vs.

```rb
[1] pry(main)> class Foo; end
=> nil
[2] pry(main)> Foo.new.bar
NoMethodError: undefined method `bar' for #<Foo:0x007ffa38031300>
from (pry):2:in `<main>'
```

Reply:
I think probably JS is the only one that won't care about accessing an attribute that doesn't exist?